src/components/time/atoms/Events.js

```diff
@@ -24,7 +23,7 @@ function renderDot(event, styles, props) {
   );
   return (
     <g
-      key={hash(event)}
+      key={event.id}
```

Comment:
Note: this part is a bit suspicious and I want to get confirmation that it's safe.

* `renderBar` and others don't have a key at all, but they should for parity.
* What happens when `evShadows` is not empty? Do we get several dots for the same event? **If yes, would these "shadow" events have some IDs?** If yes, we need to fix that first before using `event.id` everywhere.

Reply:
Yes, you're correct that when `evShadows` is not empty, the intention is to create multiple dots for the same event. Take for instance the following screenshot:

![image](https://user-images.githubusercontent.com/16495856/161625861-59631f5f-48a3-425b-a479-169ab3cb3b03.png)

You can see that many events bridge multiple categories (represented by default as horizontal lines in the timeline), and have 1-4 "shadows".

Reply:
Is there a good way to repro this scenario? Will I get it with the BLM dataset?

Reply:
Yes this screenshot is from the BLM dataset snapshot I sent. I'll just confirm that it runs that codepath, but I'm fairly sure it does.

Reply:
In fact it turns out that the BLM dataset is curated to avoid this scenario (each event has one and only one category). You can reproduce it by changing any of the associations in the "export_associations" tab of the BLM dataset from `FILTER` to `CATEGORY`, ie:

![image](https://user-images.githubusercontent.com/16495856/161628575-54c28e5a-4601-438f-bb14-01bdd18bf9c7.png)

Reply:
(Followed by saving the ODS file, hitting http://localhost:4040/api/update, and refreshing the timemap local instance.)

Reply:
OK, I'll follow up to test it and will probably need to adjust code as a result.